From 740494e3406346b277e786e686ee2e11af7909c8 Mon Sep 17 00:00:00 2001
From: John Johansen <john.johansen@canonical.com>
Date: Tue, 22 Aug 2023 01:33:03 -0700
Subject: [PATCH 57/76] UBUNTU: SAUCE: apparmor4.0.0 [57/76]: prompt - lock
 down prompt interface

While the prompt interface is supposed to be open and permissions
handled via namespacing. There are no current users that require
it be open and there are some namespacing issues that need to
be resolved. So Lock the interface down until the namespacing
issues are all fixed.

Signed-off-by: John Johansen <john.johansen@canonical.com>
---
 security/apparmor/apparmorfs.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/security/apparmor/apparmorfs.c b/security/apparmor/apparmorfs.c
index 230c4574169e..660bc558d76b 100644
--- a/security/apparmor/apparmorfs.c
+++ b/security/apparmor/apparmorfs.c
@@ -616,6 +616,8 @@ static int listener_release(struct inode *inode, struct file *file)
 {
 	struct aa_listener *listener = file->private_data;
 
+	if (!aa_current_policy_admin_capable(NULL))
+		return -EPERM;
 	if (listener)
 		aa_put_listener(listener);
 
@@ -626,6 +628,8 @@ static int listener_open(struct inode *inode, struct file *file)
 {
 	struct aa_listener *listener;
 
+	if (!aa_current_policy_admin_capable(NULL))
+		return -EPERM;
 	listener = aa_new_listener(NULL, GFP_KERNEL);
 	if (!listener)
 		return -ENOMEM;
@@ -726,6 +730,8 @@ static long listener_ioctl(struct file *file, unsigned int cmd,
 {
 	struct aa_listener *listener = file->private_data;
 
+	if (!aa_current_policy_admin_capable(NULL))
+		return -EPERM;
 	if (!listener)
 		return -EINVAL;
 
@@ -752,6 +758,9 @@ static __poll_t listener_poll(struct file *file, poll_table *pt)
 	struct aa_listener *listener = file->private_data;
 	__poll_t mask = 0;
 
+	if (!aa_current_policy_admin_capable(NULL))
+		return -EPERM;
+
 	if (listener) {
 		spin_lock(&listener->lock);
 		poll_wait(file, &listener->wait, pt);
-- 
2.42.0

